### ISSUE TITLE
Extend index-cardinality-detection to also cover hypothetical planned indexes

### ISSUE THREAD

**Issue by vin-yu** [awslabs/amazon-documentdb-tools#101]
Extend index-cardinality-detection to determine if an index is high cardinality based on an index input.

* Developers and admins planning to build indexes can use index-cardinality-detection tool to determine if the index is high cardinality. 
* This could also be used in CI/CD check point before an index change goes through into prod

**Comment by rawalpuneet** [comment]
Good idea Vin. shouldn't be a big change. I plan to call hypothetical index, planned indexes. will update the docs for that too.

**Comment by rawalpuneet** [comment]
@vin-yu / @tmcallaghan ,  can you review and see if the changes make sense to you in PR?
